substratevm/src/com.oracle.svm.driver/src/com/oracle/svm/driver/NativeImage.java

```diff
@@ -1203,11 +1204,30 @@ protected int buildImage(List<String> javaArgs, LinkedHashSet<Path> bcp, LinkedH
         }
 
         int exitStatus = 1;
+        File argsFile = null;
         try {
+            if (OS.getCurrent() == OS.WINDOWS) {
+                int totalLength = command.stream().mapToInt(s -> s.length() + 1).sum();
+                if (totalLength >= WINDOWS_MAX_COMMAND_LENGTH) {
+                    argsFile = File.createTempFile("native-image", "args");
+                    argsFile.deleteOnExit();
```

Comment:
Do we really need `.deleteOnExit()` here if we already have `argsFile.delete();` in the finally-clause?

Reply:
During testing I found out that hitting CTRL+C would not delete the file unless I also added the `.deleteOnExit()`. Now we could of course remove the `finally` block, because the process will never be long-lived, but it  seemed "untidy".

Reply:
> Now we could of course remove the finally block, because the process will never be long-lived, but it seemed "untidy".

Would be fine for me.

Reply:
Ok, done